runbotics/runbotics-desktop/src/action/file/file.action-handler.ts

```diff
@@ -114,3 +129,28 @@ export default class FileActionHandler extends StatelessActionHandler {
         }
     }
 }
+
+async function createNewFile(path: string){
+    try {
+        fs.closeSync(fs.openSync(`${path}`, 'a'));
+    } catch (err) {
+        throw Error(err);
+    }
+}
+
+function getUniqueFileName(filePath) {
+    const dir = path.dirname(filePath);
+    const ext = path.extname(filePath);
+    const baseName = path.basename(filePath, ext).replace(/\s\(\d+\)/, '');
+    
+    let counter = 1;
```

Comment:
avoid modifiable variables (so like let instead const constructors)
do not use while loop - too risky

Reply:
imo instead of counting from 1 to infinity and checking file name each time - 1. get all file names 2. get the numbers from filenames with e.g. regex or splitting by '(' and getting last element  3. find the max number 4. create new file with a number that is one number higher